dunecore/Geometry/CRPWireReadoutGeom.cxx

```diff
@@ -952,5 +958,59 @@ std::string geo::CRPWireReadoutGeom::PlaneTypeName(PlaneType_t planeType) {
   
 } // geo::CRPWireReadoutGeom::PlaneTypeName()
 
-
 // ----------------------------------------------------------------------------
+bool geo::CRPWireReadoutGeom::IsValidOpChannel(unsigned int opChannel, unsigned int NOpDets) const {
+   return fPDMapTool->isValidHardwareChannel(opChannel);
+}
+bool geo::CRPWireReadoutGeom::IsValidOpChannel(int opChannel) const {
+   return fPDMapTool->isValidHardwareChannel(opChannel);
+}
+
+//----------------------------------------------------------------------------
+unsigned int geo::CRPWireReadoutGeom::NOpChannels(unsigned int NOpDets) const
+{
+  // By default just return the number of optical channels (hardware channels)
+  return fPDMapTool->NOpChannels();
+}
+
+//----------------------------------------------------------------------------
+unsigned int geo::CRPWireReadoutGeom::MaxOpChannel(unsigned int NOpDets) const
+{
+  // By default just return the number of optical channels
+  return fPDMapTool->NOpChannels();
+}
+
+//----------------------------------------------------------------------------
+unsigned int geo::CRPWireReadoutGeom::NOpHardwareChannels(unsigned int opDet) const
+{
+  // Number of hardware channels per opDet
+  return fPDMapTool->NOpHardwareChannels(opDet);
+}
+
+//----------------------------------------------------------------------------
+unsigned int geo::CRPWireReadoutGeom::OpChannel(unsigned int detNum, unsigned int ch) const
+{
+  //What is this?
```

Comment:
From the json file it seems this ch is not something meaningful

Reply:
I'm overwriting all the PDS functions in WireRadout.
There is a logic behind related with the number of cryostat in the geometry, but I believe this do not affect us in PDVD.